src/vm/hints/cairo_keccak/keccak_hints.rs

```diff
@@ -0,0 +1,414 @@
+            KECCAK_STATE_SIZE_FELTS,
+        ));
+    }
+
+    let values = vm
+        .memory
+        .get_range(
+            &MaybeRelocatable::RelocatableValue(keccak_ptr.sub(KECCAK_STATE_SIZE_FELTS)?),
+            KECCAK_STATE_SIZE_FELTS,
+        )
+        .map_err(VirtualMachineError::MemoryError)?;
+
+    let mut u64_values = maybe_reloc_vec_to_u64_array(&values)?;
+
+    // this function of the keccak crate is the one used instead of keccak_func from
+    // keccak_utils.py
+    keccak::f1600(&mut u64_values);
+
+    let bigint_values = u64_array_to_bigint_vec(&u64_values);
+
+    vm.segments
+        .write_arg(&mut vm.memory, &keccak_ptr, &bigint_values, Some(&vm.prime))
+        .map_err(VirtualMachineError::MemoryError)?;
+
+    Ok(())
+}
+
+/*
+    %{
+        # Add dummy pairs of input and output.
+        _keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)
+        _block_size = int(ids.BLOCK_SIZE)
+        assert 0 <= _keccak_state_size_felts < 100
+        assert 0 <= _block_size < 10
+        inp = [0] * _keccak_state_size_felts
+        padding = (inp + keccak_func(inp)) * _block_size
+        segments.write_arg(ids.keccak_ptr_end, padding)
+    %}
+*/
+
+pub fn cairo_keccak_finalize(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    if KECCAK_STATE_SIZE_FELTS >= 100 {
+        return Err(VirtualMachineError::InvalidKeccakStateSizeFelts(
+            KECCAK_STATE_SIZE_FELTS,
+        ));
+    }
+
+    if BLOCK_SIZE >= 10 {
+        return Err(VirtualMachineError::InvalidBlockSize(BLOCK_SIZE));
+    }
+
+    let mut inp = [0u64; KECCAK_STATE_SIZE_FELTS];
+    keccak::f1600(&mut inp);
+
+    let mut padding = vec![bigint!(0_u64); KECCAK_STATE_SIZE_FELTS];
+    padding.extend(u64_array_to_bigint_vec(&inp));
+
+    let base_padding = padding.clone();
+
+    for _ in 0..(BLOCK_SIZE - 1) {
+        padding.extend(base_padding.clone());
+    }
+
+    let keccak_ptr_end = get_ptr_from_var_name(
+        "keccak_ptr_end",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    vm.segments
+        .write_arg(&mut vm.memory, &keccak_ptr_end, &padding, Some(&vm.prime))
+        .map_err(VirtualMachineError::MemoryError)?;
+
+    Ok(())
+}
+
+// Helper function to transform a vector of MaybeRelocatables into a vector
+// of u64. Raises error if there are None's or if MaybeRelocatables are not Bigints.
+fn maybe_reloc_vec_to_u64_array(
+    vec: &[Option<&MaybeRelocatable>],
+) -> Result<[u64; KECCAK_STATE_SIZE_FELTS], VirtualMachineError> {
+    let array: [u64; KECCAK_STATE_SIZE_FELTS] = vec
+        .iter()
+        .map(|n| {
+            if let Some(MaybeRelocatable::Int(num)) = n {
+                num.to_u64().ok_or(VirtualMachineError::BigintToU64Fail)
+            } else {
+                Err(VirtualMachineError::NoneInMemoryRange)
```

Comment:
`ExpectedInt` would be a better error. Otherwise, a `Relocatable` would give us a misleading error.